Gems/ROS2/Code/Source/Frame/ROS2FrameComponent.cpp

```diff
@@ -73,6 +73,29 @@ namespace ROS2
                 GetFrameID().data(),
                 IsDynamic() ? "continuously to /tf" : "once to /tf_static");
 
+            if (!IsDynamic())
+            {
+                bool hasMovableJoints = false;
+                const auto& components = this->m_entity->GetComponents();
+                for (const auto component : components)
+                {
+                    const auto type = component->GetUnderlyingComponentType();
+
+                    if (type == AZ::Uuid("{9C5B955C-6C80-45FA-855D-DDA449C85313}") || // PhysXHingeJoint
+                        type == AZ::Uuid("{CEE6A6DF-FDE1-4E30-9EE2-631C7561C1C7}") || // PhysXPrismaticJoint
+                        type == AZ::Uuid("{A5CA0031-72E4-4908-B764-EDECD3091882}") || // HingeJointComponent
+                        type == AZ::Uuid("{9B34CA1B-C063-4D42-A15B-CE6CD7C828DC}")) // PrismaticJointComponent
+                    {
+                        hasMovableJoints = true;
+                        break;
+                    }
+                }
+                AZ_Warning(
```

Comment:
This warning could be removed if the value m_isDynamic is simply set by the logic and not settable by the user